library/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumFeature.kt

```diff
@@ -768,6 +771,8 @@ class RumFeature internal constructor(
         internal const val DD_TELEMETRY_CONFIG_SAMPLE_RATE_TAG =
             "_dd.telemetry.configuration_sample_rate"
 
+        internal val DEFAULT_NOOP_SDK_CORE = NoOpSdkCore()
```

Comment:
I have a bit of concern regarding using `NoOpSdkCore` in features (and making it public). Normally features should respect lifecycle, so when we provide `NoOpSdkCore` if things go wrong they won't be noticed (because there is no logger behind).

But the other alternative will force us to either use `lateinit` or will add nullability. But at least if things go wrong we will get an exception (which normally we should get during our instrumentation/nightly tests, but there is a risk of exposure on the client side).

wdyt?